core/jazz_codeq/index.js

```diff
@@ -30,15 +30,22 @@ function handler(event, context, cb) {
 	let config = configModule.getConfig(event, context);
 
 	logger.init(serviceContext, context);
-	
+
 	try {
 		logger.debug(serviceContext.resourcePath);
 		utils.getAPIPath(serviceContext.resourcePath)
 			.then(data => {
-				const pathString = data.pathString;
+        const pathString = data.pathString;
+        let serviceId
 				logger.debug(pathString);
 				if (serviceContext && serviceContext.method && serviceContext.method === 'GET' && pathString === "codeq") {
 					logger.debug(`code quality service called with pathstring - ${pathString}`);
+          let header_key = config.SERVICE_ID_HEADER_KEY.toLowerCase();
+          if (!serviceContext.headers[header_key]) {
```

Comment:
Can you check if other core services (that require serviceId) work in a similar fashion? Check if key exists in header..